modules/internal/DefaultRectangular.chpl

```diff
@@ -1186,25 +1355,14 @@ module DefaultRectangular {
           // If we detect that blk is never changed then then blk(rank) == 1.
           // Knowing this, we need not multiply the final ind(...) by anything.
           // This relies on us marking every function that modifies blk
-          if __primitive("optimize_array_blk_mult") {
```

Comment:
Good catch on the comment.  Now you know why I rarely write comments -- It seems I rarely read them either.  :)

Tacking onto Elliot's note:  While it was tempting to pull out all Elliot's inner-mult-optimization code in this PR, we decided to wait for a separate PR for that (I pulled the tests of his optimization out because they became harder to maintain than seemed worthwhile).